probdists/Triangulardistribution.py

```diff
@@ -0,0 +1,199 @@
+
+    def calculate_mean(self, round_to=2):
+        """
+        Method to calculate the mean from the min, max and mode
+
+        Args:
+            round_to (int): Round the mean value. Defaults to 2.
+
+        Returns:
+            float: mean of the data set
+        """
+
+        self.mean = 1/3 * (self.a + self.b + self.mode)
+
+        return round(self.mean, round_to)
+
+    def calculate_stdev(self, round_to=2):
+        """
+        Method to calculate the standard deviation from the min, max and mode
+
+        Args:
+            round_to (int): Round the mean value. Defaults to 2.
+
+        Returns:
+            float: standard deviation of the data set
+        """
+
+        summation = (self.a ** 2) + (self.b ** 2) + (self.mode ** 2) - \
+                    (self.a * self.b) - (self.a * self.mode) - \
+                    (self.b * self.mode)
+        variance = summation / 18
+        self.stdev = math.sqrt(variance)
+
+        return round(self.stdev, round_to)
+
+    def replace_stats_with_data(self):
+        """ Method to calculate a, b, mode from the data set
+
+        Args:
+            None
+
+        Returns:
+            float: a, the minimum value
+            float: b, the maximum value
+            float: mode, the mode of the dataset
+        """
+        if not self.data:
+            # Use default values
+            self.a, self.b, self.mode = 0, 1, 0.5
+        else:
+            self.a = min(self.data)
+            self.b = max(self.data)
+            self.calculate_mode()
+
+        return self.a, self.b, self.mode
+
+    def calculate_mode(self, round_to=2):
+        """
+        Calculates the mode of a dataset
+        If no single mode, it will approximate the mode using the mean
+
+        Args:
+            round_to (int): Round the mode value. [Default value: 2]
+
+        Returns:
+            float: mode of data
+        """
+        data = Counter(self.data)
+        dict_data = dict(data)
+
+        # Create list of modes from data
+        mode = [k for k, v in dict_data.items() if v == max(list(data.values()))]
+
+        if len(mode) == 1:
+            self.mode = mode[0]
+        else:
+            # Multiple modes
+            # Approximates mode using mean
+            self.mode = sum(self.data) / len(self.data)
+        return round(self.mode, round_to)
+
+    def calculate_pdf(self, x, round_to=2):
+        """
+        Probability density function calculator for the Triangular distribution.
+
+        Args:
+            x (float): point for calculating the probability density function
+            round_to (int): Round the pdf value. [Default value: 2]
+
+        Returns:
+            float: probability density function
+        """
+        value = 0  # default value for when x < min or x > max
+        if self.a <= x <= self.mode:
+            value = (2 * (x - self.a)) / (
+                    (self.b - self.a) * (self.mode - self.a))
```

Comment:
how do we plan to handle division by zero here?

Reply:
I think the best way would be to check for if a == b during instantiation, and raise a ValueError because that would suggest the data is the same number since a is minimum and b is maximum. Then in any calculations, it should be safe to assume that a != b != mode
This applies to the other comments about ZeroDivision cases

Reply:
Yes, that sounds like a good idea!